op-conductor/consensus/raft.go

```diff
@@ -86,18 +104,31 @@ func NewRaftConsensus(log log.Logger, cfg *RaftConsensusConfig) (*RaftConsensus,
 		return nil, fmt.Errorf(`raft.NewFileSnapshotStore(%q): %w`, baseDir, err)
 	}
 
-	addr, err := net.ResolveTCPAddr("tcp", cfg.ServerAddr)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to resolve tcp address")
+	var advertiseAddr net.Addr
+	if cfg.AdvertisedAddr == "" {
+		log.Warn("No advertised address specified. Advertising local address.")
```

Comment:
[nit] should this be a warn, or is it expected in some cases?

Reply:
During tests it's expected to not advertise a pre-configured address, but during regular operation I imagine the address it binds to might not be the right one to advertise.

 Maybe if you configure it with `--consensus.addr` to bind to the exact IP that it listens on it still advertises correctly, but if you listen to `127.0.0.1` or `0.0.0.0` it maybe wouldn't advertise the right thing if it's on the same system I think. I don't know if I understand the advertisement functionality well enough to say whether it matters. Previous behavior did seem to only care about advertising the same address it listens on though. And since there's an RPC on op-conductor to do connections and such later, maybe it doesn't matter at all what is configured as the advertised addr.

 In any case, I think it's worth a warning if an implicit address/port is used.

Reply:
Actually this is helpful, would prefer to listen on 0.0.0.0 and advertise the server address separately